src/helm/benchmark/metrics/classification_metrics.py

```diff
@@ -68,3 +70,38 @@ def evaluate_instances(self, request_states: List[RequestState]) -> List[Stat]:
             Stat(MetricName("classification_macro_f1")).add(f1_score(y_pred=y_pred, y_true=y_true, average="macro")),
             Stat(MetricName("classification_micro_f1")).add(f1_score(y_pred=y_pred, y_true=y_true, average="micro")),
         ]
+
+
+class CLEVAMultipleChoiceClassificationMetric(Metric):
+    """
+    Calculate population micro/macro F1 score for CLEVA benchmark.
+    Intends to works for multiple_choice_* adapters.
+    """
+
+    def evaluate_instances(self, request_states: List[RequestState]) -> List[Stat]:
+        y_pred: List[str] = []
+        y_true: List[str] = []
+        for request_state in request_states:  # one request state per instance
+            if request_state.request_mode == "calibration":
+                raise ValueError("ClassificationMetric does not support calibration requests")
+            golds: List[Reference] = [
+                reference for reference in request_state.instance.references if reference.is_correct
+            ]
+            assert len(golds) > 0, "CLEVAMultipleChoiceClassificationMetric are designed for multiple_choice_* adapters"
+            assert request_state.result is not None
+            sorted_completions: List[Sequence] = sorted(request_state.result.completions, key=lambda x: -x.logprob)
```

Comment:
Are you sure that logprob is available in the `multiple_choice_joint` adapter?

Reply:
Yes, it works fine. The methodology of my implementation is practically the same as BasicMetric.compute_reference_metrics(). As long as one can use basic_metrics, my classification_f1 should work as well.